Automated publish fails using Node 16

Saw this error during the auto-publish:

```
> ./test/browser/generate-index.sh && NODE_OPTIONS=--openssl-legacy-provider npm run build:browser && npm run build:docs

/home/runner/work/opossum/opossum
Processing cache.js
Processing circuit-shutdown-test.js
Processing closed-test.js
Processing common.js
Processing context-test.js
Processing enable-disable-test.js
Processing error-filter-test.js
Processing half-open-test.js
Processing health-check-test.js
Processing semaphore-test.js
Processing state-test.js
Processing status-test.js
Processing test.js
Processing volume-threshold-test.js
Processing warmup-test.js
node: --openssl-legacy-provider is not allowed in NODE_OPTIONS
npm ERR! code 9
npm ERR! path /home/runner/work/opossum/opossum
npm ERR! command failed
npm ERR! command sh -c -- npm run build

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2023-05-25T14_33_[47](https://github.com/nodeshift/opossum/actions/runs/5081204702/jobs/9129221168#step:7:48)_623Z-debug-0.log
Error: Process completed with exit code 9.
```


This also fails locally with node 16, i changed to node 18 and it worked.

So we should just be able to update the GH action to use node 18

--- a/.github/workflows/release-please.yml
+++ b/.github/workflows/release-please.yml
@@ -18,7 +18,7 @@ jobs:
         if: ${{ steps.release.outputs.release_created }}
       - uses: actions/setup-node@v3
         with:
-          node-version: 16
+          node-version: 18
           registry-url: 'https://registry.npmjs.org'
         if: ${{ steps.release.outputs.release_created }}
       - run: npm ci

